internal/services/firewall/firewall_policy_application_rule_collection_resource.go

```diff
@@ -0,0 +1,432 @@
+	}
+
+	param := network.FirewallPolicyRuleCollectionGroup{
+		FirewallPolicyRuleCollectionGroupProperties: &network.FirewallPolicyRuleCollectionGroupProperties{
+			Priority:        resp.Priority,
+			RuleCollections: resp.RuleCollections,
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, ruleCollectionGroupId.RuleCollectionGroupName, param)
+	if err != nil {
+		return fmt.Errorf("creating Firewall Policy Rule Collection %q (Resource Group %q / Policy: %q / Rule Collection Group: %q): %+v", name, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, ruleCollectionGroupId.RuleCollectionGroupName, err)
+	}
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting Firewall Policy Rule Collection Group %q (Resource Group %q / Policy: %q / Rule Collection Group: %q): %+v", name, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, ruleCollectionGroupId.RuleCollectionGroupName, err)
+	}
+
+	d.SetId(parse.NewFirewallPolicyRuleCollectionID(ruleCollectionGroupId.SubscriptionId, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, ruleCollectionGroupId.RuleCollectionGroupName, name).ID())
+
+	return resourceFirewallPolicyApplicationRuleCollectionRead(d, meta)
+}
+
+func resourceFirewallPolicyApplicationRuleCollectionRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Firewall.FirewallPolicyRuleGroupClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FirewallPolicyRuleCollectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.FirewallPolicyName, id.RuleCollectionGroupName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Firewall Policy Rule Collection Group %q was not found in Resource Group %q - removing from state!", id.RuleCollectionGroupName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Firewall Policy Rule Collection Group %q (Resource Group %q / Policy: %q): %+v", id.RuleCollectionGroupName, id.ResourceGroup, id.FirewallPolicyName, err)
+	}
+
+	idx := indexFunc(*resp.RuleCollections, func(policy network.BasicFirewallPolicyRuleCollection) bool {
+		info, _ := policy.AsFirewallPolicyFilterRuleCollection()
+		return *info.Name == id.RuleCollectionName
+	})
+
+	if idx == -1 {
+		log.Printf("[DEBUG] Firewall Policy Rule Collection %q was not found in Firewall Policy Rule Collection Group %q - removing from state!", id.RuleCollectionName, id.RuleCollectionGroupName)
+		d.SetId("")
+		return nil
+	}
+
+	ruleCollection, _ := (*resp.RuleCollections)[idx].AsFirewallPolicyFilterRuleCollection()
+
+	d.Set("name", ruleCollection.Name)
+	d.Set("action", ruleCollection.Action.Type)
```

Comment:
Nil check on `.Action` needed, if it is nil, set an empty string to the `action`.